packages/safe-tools-client/app/services/hub-authentication.ts

```diff
@@ -48,27 +49,38 @@ export default class HubAuthenticationService extends Service {
     return await getSDK('HubAuth', ethersProvider, config.hubUrl);
   }
 
+  get authTokenKey() {
+    return `authToken-${this.wallet.address}-${this.network.chainId}`;
+  }
+
   get authToken(): string | null {
-    return this.storage.getItem('authToken');
+    return this.storage.getItem(this.authTokenKey);
   }
 
   set authToken(val: string | null) {
     if (val) {
-      this.storage.setItem('authToken', val);
+      this.storage.setItem(this.authTokenKey, val);
     } else {
       this.isAuthenticated = false;
-      this.storage.removeItem('authToken');
+      this.storage.removeItem(this.authTokenKey);
     }
   }
 
   async hasValidAuthentication() {
     const hubAuth = await this.getHubAuth();
-
-    return Boolean(
-      this.wallet.isConnected &&
-        this.authToken &&
-        (await hubAuth.checkValidAuth(this.authToken))
-    );
+    try {
+      const authAddress = this.authToken
+        ? await hubAuth.getAddress(this.authToken)
+        : null;
+      return Boolean(
+        this.wallet.isConnected && authAddress === this.wallet.address
+      );
+    } catch (e) {
+      if (e === 'Invalid auth token') {
```

Comment:
Isn't `e` an `Error` type? Did you mean to use `e.message`?

Reply:
fixed